libs/langchain/langchain/agents/output_parsers/openai_functions.py

```diff
@@ -14,7 +15,48 @@
 from langchain.agents.agent import AgentOutputParser
 
 
-class OpenAIFunctionsAgentOutputParser(AgentOutputParser):
+class BaseFunctionsAgentOutputParser(AgentOutputParser):
```

Comment:
I like creating a `BaseFunctionsAgentOutputParser` that both of these fit into, but I think this class and the `VertexAIFunctionsAgentOutputParser` should be in a separate file outside of the OpenAI parser